client/js/table-view.js

```diff
@@ -2,119 +2,163 @@ const { getLetterRange } = require('./array-util');
-				}
-			}
-			fragment.appendChild(tr);
-		}
-		removeChildren(this.sheetBodyEl);
-		this.sheetBodyEl.appendChild(fragment);
-		this.renderSumRow();
-	}
-
-	renderSumRow() {
-		const tr = createTR();
-		for(let col = 0; col < this.model.numCols; col++) {
-			const total = this.calculateColTotal(col);
-			const td = createTD(total);
-			td.className = 'sum-cell';
-			tr.appendChild(td);
-		}
-		this.sheetBodyEl.appendChild(tr);
-	}
-
-	calculateColTotal(col) {
-		let total = 0;
-		for(let row = 0; row < this.model.numRows; row++) {
-			const position = {col: col, row: row};
-			total += this.normalizeValueForSumming(this.model.getValue(position));
-		}
-		console.log(total);
-		return total;
-	}
-
-	normalizeValueForSumming(value) {
-		const intVal = parseInt(value);
-		return intVal || 0;
-	}
-
-	attachEventHandlers() {
-		this.sheetBodyEl.addEventListener('click', 
-			this.handleSheetClick.bind(this));
-		this.formulaBarEl.addEventListener('keyup', this.handleFormulaBarChange.bind(this));
-	}
-
-
-	handleSheetClick(evt){
-		const col = evt.target.cellIndex;
-		const row = evt.target.parentElement.rowIndex - 1;
-		this.currentCellLocation = {col: col, row: row};
-		this.renderTableBody();
-		this.renderFomulaBar();
-	}
+  constructor(model) {
+    this.model = model;
+  }
+  init() {
+    this.initDomReferences();
+    this.initCurrentCell();
+    this.renderTable();
+    this.attachEventHandlers();
+  }
+  initDomReferences() {
+    this.headerRowEl = document.querySelector('THEAD TR');
+    this.sheetBodyEl = document.querySelector('TBODY');
+    this.formulaBarEl = document.querySelector('#formula-bar');
+  }
+
+  initCurrentCell() {
+    this.currentCellLocation = {col: 1, row: 0};
+    this.renderFomulaBar();
+  }
+
+  normalizeValueForRendering(value) {
+    return value || '';
+  }
+
+  renderFomulaBar() {
+    const currentCellValue = this.model.getValue(this.currentCellLocation);
+    this.formulaBarEl.value = this.normalizeValueForRendering(currentCellValue);
+    this.formulaBarEl.focus();
+  }
+
+  handleFormulaBarChange(evt) {
+    const value = this.formulaBarEl.value;
+    this.model.setValue(this.currentCellLocation, value);
+    this.renderTableBody();
+  }
+  renderTable() {
+    this.renderTableHeader();
+    this.renderTableBody();
+  }
+  renderTableHeader() {
+    //clear header row
+    removeChildren(this.headerRowEl);
+    //get letters and build elements
+    getLetterRange('A', this.model.numCols)
+      .map(colLabel => createTH(colLabel))
+      .forEach(th => this.headerRowEl.appendChild(th));
+  }
+  isCurrentRow(row) {
+    return  this.currentCellLocation.col == 0 &&
+           this.currentCellLocation.row == row;
+  }
+  isCurrentCol(col) {
+    return this.currentColLocation == col
+  }
+
+  isCurrentCell(col,row) {
+    return this.currentCellLocation.col == col &&
+           this.currentCellLocation.row == row
+  }
+
+  renderTableBody(){
+    const fragment = document.createDocumentFragment();
```

Comment:
This ended up being a pretty long function that does a few different things. Can you separate into helper functions? For instance, one helper function could be "nameTD" and it could do the class name selection process.